compat/cpu.h

```diff
@@ -9,9 +9,12 @@
 #ifndef SQUID_COMPAT_CPU_H
 #define SQUID_COMPAT_CPU_H
 
-#if HAVE_ERRNO_H
-#include <errno.h> /* for ENOTSUP */
+#ifndef __cplusplus
+#error "This header is only meant to be used from c++ source files"
 #endif
+
+#include <cerrno>
```

Comment:
Can you give me an example where the `#error` would affect the compilation outcome? I would expect the C compiler to fail on `#include <cerrno>`. If it fails, then I do not think we should bother with  such `#error` clauses. There is just not enough gain from them in that case...

> At https://github.com/squid-cache/squid/pull/1314#issuecomment-1513733817: "if it's in compat/, let's use C headers until we have C files".

IMO, to be consistent, we would then need to prohibit C++ code from `compat/`, which is doable but would make things worse overall (and move us in the opposite direction of the "no C files" goal).

Still, I am not vetoing this idea if you really like it and can come up with a plausible example to address the concern at the beginning of this change request.

Reply:
That is a fair point. Removing this